### PR TITLE
docs: Add m1 specific build instructions (#264)

### LINKED ISSUE #264: Document development instructions on Apple Silicon / M1
We need to add a section in middle to specify the Apple M1 specific build instructions and the workarounds we had to do to get it working on M1. 

Document to edit:
https://github.com/srevinsaju/guiscrcpy/blob/main/docs/BUILD.md

### PR DESCRIPTION
**Pull request by gouravkhunger**
Documentation for Apple Silicon chips added along with a video for better understanding 🎉
Resolve #264

### CODE REVIEW

**Comment by srevinsaju** [comment]
Thanks, I will review it today :rocket:

**Comment by gouravkhunger** [comment]
Sure 🚀 
I will be messing around with homebrew and soon figure out the publishing process 😄

File: docs/BUILD.md [hunk 1]
```diff
@@ -49,8 +49,31 @@ pip3 install .
 
 Use `--user` parameter if necessary.
 
+### M1 specific build instructions
 
+Macs with Apple Silicon chips need to follow these steps to build. Watch [this video](https://youtu.be/JsqV5QJQ0Y4) for aid.
```
Comment (srevinsaju) [inline docs/BUILD.md:54]: Nice video 👌

**Comment by srevinsaju** [comment]
Great work! reviewed :smile:

**Review by srevinsaju — APPROVED** [review]

**Comment by gouravkhunger** [comment]
Thank you 💯

**Pull request merged**